packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -284,7 +296,24 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
 
   void _handlePersistentFrameCallback(Duration timeStamp) {
     drawFrame();
-    _mouseTracker.schedulePostFrameCheck();
+    _scheduleMouseTrackerUpdate();
+  }
+
+  bool _debugMouseTrackerUpdateScheduled = false;
+  void _scheduleMouseTrackerUpdate() {
```

Comment:
Couldn't you leave `_mouseTracker.schedulePostFrameCheck()`, and just replace the body of that function with the contents of this one?

I.e., doesn't this code belong in the mouse tracker, and not the binding?

Reply:
I think the idea is that `MouseTracker` does not schedule for itself. It provides a method that should be scheduled, and the binding schedule things, just like how the binding subscribes to the pointer router. In this way, `MouseTracker` only provides 2 methods that is called outside, and is agnostic by whom, when, and how often they are called.

Reply:
OK, makes sense.